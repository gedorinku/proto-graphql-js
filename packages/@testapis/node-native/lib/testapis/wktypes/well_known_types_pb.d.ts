// package: testapis.wktypes
// file: testapis/wktypes/well_known_types.proto

import * as jspb from "google-protobuf";
import * as google_protobuf_timestamp_pb from "google-protobuf/google/protobuf/timestamp_pb";
import * as google_protobuf_wrappers_pb from "google-protobuf/google/protobuf/wrappers_pb";

export class Message extends jspb.Message {
  hasTimestamp(): boolean;
  clearTimestamp(): void;
  getTimestamp(): google_protobuf_timestamp_pb.Timestamp | undefined;
  setTimestamp(value?: google_protobuf_timestamp_pb.Timestamp): void;

  hasInt32Value(): boolean;
  clearInt32Value(): void;
  getInt32Value(): google_protobuf_wrappers_pb.Int32Value | undefined;
  setInt32Value(value?: google_protobuf_wrappers_pb.Int32Value): void;

  hasInt64Value(): boolean;
  clearInt64Value(): void;
  getInt64Value(): google_protobuf_wrappers_pb.Int64Value | undefined;
  setInt64Value(value?: google_protobuf_wrappers_pb.Int64Value): void;

  hasUint32Value(): boolean;
  clearUint32Value(): void;
  getUint32Value(): google_protobuf_wrappers_pb.UInt32Value | undefined;
  setUint32Value(value?: google_protobuf_wrappers_pb.UInt32Value): void;

  hasUint64Value(): boolean;
  clearUint64Value(): void;
  getUint64Value(): google_protobuf_wrappers_pb.UInt64Value | undefined;
  setUint64Value(value?: google_protobuf_wrappers_pb.UInt64Value): void;

  hasFloatValue(): boolean;
  clearFloatValue(): void;
  getFloatValue(): google_protobuf_wrappers_pb.FloatValue | undefined;
  setFloatValue(value?: google_protobuf_wrappers_pb.FloatValue): void;

  hasDoubleValue(): boolean;
  clearDoubleValue(): void;
  getDoubleValue(): google_protobuf_wrappers_pb.DoubleValue | undefined;
  setDoubleValue(value?: google_protobuf_wrappers_pb.DoubleValue): void;

  hasBoolValue(): boolean;
  clearBoolValue(): void;
  getBoolValue(): google_protobuf_wrappers_pb.BoolValue | undefined;
  setBoolValue(value?: google_protobuf_wrappers_pb.BoolValue): void;

  hasStringValue(): boolean;
  clearStringValue(): void;
  getStringValue(): google_protobuf_wrappers_pb.StringValue | undefined;
  setStringValue(value?: google_protobuf_wrappers_pb.StringValue): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): Message.AsObject;
  static toObject(includeInstance: boolean, msg: Message): Message.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: Message, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): Message;
  static deserializeBinaryFromReader(message: Message, reader: jspb.BinaryReader): Message;
}

export namespace Message {
  export type AsObject = {
    timestamp?: google_protobuf_timestamp_pb.Timestamp.AsObject,
    int32Value?: google_protobuf_wrappers_pb.Int32Value.AsObject,
    int64Value?: google_protobuf_wrappers_pb.Int64Value.AsObject,
    uint32Value?: google_protobuf_wrappers_pb.UInt32Value.AsObject,
    uint64Value?: google_protobuf_wrappers_pb.UInt64Value.AsObject,
    floatValue?: google_protobuf_wrappers_pb.FloatValue.AsObject,
    doubleValue?: google_protobuf_wrappers_pb.DoubleValue.AsObject,
    boolValue?: google_protobuf_wrappers_pb.BoolValue.AsObject,
    stringValue?: google_protobuf_wrappers_pb.StringValue.AsObject,
  }
}

