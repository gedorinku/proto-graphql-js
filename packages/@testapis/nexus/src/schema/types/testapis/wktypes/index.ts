export * from "./well_known_types_pb_nexus";
