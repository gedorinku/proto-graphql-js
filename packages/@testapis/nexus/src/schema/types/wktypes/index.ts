export * from "./well_known_types_nexus_pb";
