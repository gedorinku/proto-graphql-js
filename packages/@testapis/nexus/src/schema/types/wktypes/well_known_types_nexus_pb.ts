// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: wktypes/well_known_types.proto

import { objectType } from "@nexus/schema";
import * as proto_nexus from "proto-nexus";
export const Message = objectType({
    name: "Message",
    description: "",
    definition(t) {
        t.nullable.string("timestamp", {
            description: "",
            resolve(root) { return root.getTimestamp(); },
        });
        t.nullable.int("int32Value", {
            description: "",
            resolve(root) { return proto_nexus.unwrapInt32Value(root.getInt32Value()); },
        });
        t.nullable.int("int64Value", {
            description: "",
            resolve(root) { return proto_nexus.unwrapInt64Value(root.getInt64Value()); },
        });
        t.nullable.int("uint32Value", {
            description: "",
            resolve(root) { return proto_nexus.unwrapUInt32Value(root.getUint32Value()); },
        });
        t.nullable.int("uint64Value", {
            description: "",
            resolve(root) { return proto_nexus.unwrapUInt64Value(root.getUint64Value()); },
        });
        t.nullable.float("floatValue", {
            description: "",
            resolve(root) { return proto_nexus.unwrapFloatValue(root.getFloatValue()); },
        });
        t.nullable.float("doubleValue", {
            description: "",
            resolve(root) { return proto_nexus.unwrapDoubleValue(root.getDoubleValue()); },
        });
        t.nullable.boolean("boolValue", {
            description: "",
            resolve(root) { return proto_nexus.unwrapBoolValue(root.getBoolValue()); },
        });
        t.nullable.string("stringValue", {
            description: "",
            resolve(root) { return proto_nexus.unwrapStringValue(root.getStringValue()); },
        });
    },
    rootTyping: { name: "Message", path: "@testapis/node/lib/wktypes/well_known_types_pb" },
});
