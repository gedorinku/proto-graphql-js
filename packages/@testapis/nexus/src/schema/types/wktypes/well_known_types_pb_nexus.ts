// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: wktypes/well_known_types.proto

import { objectType, inputObjectType, nullable } from "nexus";
import * as $$proto_nexus$protobufjs_adapter from "@proto-nexus/protobufjs-adapter";
import * as $$testapis$node$lib$wktypes from "@testapis/node/lib/wktypes";
export type $$testapis$node$lib$wktypes$testapi$wktypes$Message = $$testapis$node$lib$wktypes.testapi.wktypes.IMessage;
export const Message = objectType({
    name: "Message",
    description: "",
    definition(t) {
        t.field("timestamp", {
            type: nullable("DateTime"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.timestampToDate(root.timestamp); }
        });
        t.field("int32Value", {
            type: nullable("Int"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.unwrapInt32Value(root.int32Value); }
        });
        t.field("int64Value", {
            type: nullable("String"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.unwrapInt64Value(root.int64Value)?.toString() ?? null; }
        });
        t.field("uint32Value", {
            type: nullable("Int"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.unwrapUInt32Value(root.uint32Value); }
        });
        t.field("uint64Value", {
            type: nullable("String"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.unwrapUInt64Value(root.uint64Value)?.toString() ?? null; }
        });
        t.field("floatValue", {
            type: nullable("Float"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.unwrapFloatValue(root.floatValue); }
        });
        t.field("doubleValue", {
            type: nullable("Float"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.unwrapDoubleValue(root.doubleValue); }
        });
        t.field("boolValue", {
            type: nullable("Boolean"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.unwrapBoolValue(root.boolValue); }
        });
        t.field("stringValue", {
            type: nullable("String"),
            description: "",
            resolve(root) { return $$proto_nexus$protobufjs_adapter.unwrapStringValue(root.stringValue); }
        });
    },
    sourceType: { module: __filename, export: "$$testapis$node$lib$wktypes$testapi$wktypes$Message" }
});
export const MessageInput = inputObjectType({
    name: "MessageInput",
    description: "",
    definition(t) {
        t.field("timestamp", {
            type: nullable("DateTime"),
            description: ""
        });
        t.field("int32Value", {
            type: nullable("Int"),
            description: ""
        });
        t.field("int64Value", {
            type: nullable("String"),
            description: ""
        });
        t.field("uint32Value", {
            type: nullable("Int"),
            description: ""
        });
        t.field("uint64Value", {
            type: nullable("String"),
            description: ""
        });
        t.field("floatValue", {
            type: nullable("Float"),
            description: ""
        });
        t.field("doubleValue", {
            type: nullable("Float"),
            description: ""
        });
        t.field("boolValue", {
            type: nullable("Boolean"),
            description: ""
        });
        t.field("stringValue", {
            type: nullable("String"),
            description: ""
        });
    }
});
