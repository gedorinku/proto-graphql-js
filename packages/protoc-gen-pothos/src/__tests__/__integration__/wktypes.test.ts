import { testSchemaGeneration } from "../__helpers__/process.test.helper";

testSchemaGeneration("wktypes", "ts-proto", {
  builderTs: `
    import SchemaBuilder from "@pothos/core";
    import { GraphQLDateTime } from "graphql-scalars";

    export const builder = new SchemaBuilder<{
      Scalars: {
        DateTime: { Input: Date, Output: Date },
      },
    }>({});
    builder.queryType();
    builder.addScalarType("DateTime", GraphQLDateTime, {});
  `,
  schemaTests: [
    [
      "values are present",
      {
        extraSchema: `
          import { builder } from "./builder";
          import { Message as MessageRef } from "./testapis/wktypes/well_known_types.pb.pothos";
          import { Message } from "@testapis/ts-proto/lib/testapis/wktypes/well_known_types";

          builder.queryField("test", (f) => f.field({
            type: MessageRef,
            resolve() {
              return Message.fromPartial({
                timestamp: new Date(1609137725453),
                int32Value: 2,
                int64Value: "4",
                uint32Value: 5,
                uint64Value: "6",
                floatValue: 3.5,
                doubleValue: 2.4,
                boolValue: true,
                stringValue: "foobar",
              });
            },
          }));
        `,
        testQuery: `
          query Test {
            test {
              ...Message
            }
          }
          fragment Message on Message {
            timestamp
            int32Value
            int64Value
            uint32Value
            uint64Value
            floatValue
            doubleValue
            boolValue
            stringValue
          }
        `,
      },
    ],
    [
      "values are blank",
      {
        extraSchema: `
          import { builder } from "./builder";
          import { Message as MessageRef } from "./testapis/wktypes/well_known_types.pb.pothos";
          import { Message } from "@testapis/ts-proto/lib/testapis/wktypes/well_known_types";

          builder.queryField("test", (f) => f.field({
            type: MessageRef,
            resolve() {
              return Message.fromPartial({});
            },
          }));
        `,
        testQuery: `
          query Test {
            test {
              ...Message
            }
          }
          fragment Message on Message {
            timestamp
            int32Value
            int64Value
            uint32Value
            uint64Value
            floatValue
            doubleValue
            boolValue
            stringValue
          }
        `,
      },
    ],
  ],
});
