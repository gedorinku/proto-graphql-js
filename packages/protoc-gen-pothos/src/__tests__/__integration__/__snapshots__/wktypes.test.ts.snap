// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema generation with ts-proto values are blank 1`] = `
Object {
  "data": Object {
    "test": Object {
      "boolValue": null,
      "doubleValue": null,
      "floatValue": null,
      "int32Value": null,
      "int64Value": null,
      "stringValue": null,
      "timestamp": null,
      "uint32Value": null,
      "uint64Value": null,
    },
  },
}
`;

exports[`schema generation with ts-proto values are blank: schema.graphql 1`] = `
"\\"\\"\\"
A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the \`date-time\` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
\\"\\"\\"
scalar DateTime

type Message {
  boolValue: Boolean
  boolValues: [Boolean!]
  doubleValue: Float
  doubleValues: [Float!]
  floatValue: Float
  floatValues: [Float!]
  int32Value: Int
  int32Values: [Int!]
  int64Value: String
  int64Values: [String!]
  stringValue: String
  stringValues: [String!]
  timestamp: DateTime
  timestamps: [DateTime!]
  uint32Value: Int
  uint32Values: [Int!]
  uint64Value: String
  uint64Values: [String!]
}

input MessageInput {
  boolValue: Boolean
  boolValues: [Boolean!]
  doubleValue: Float
  doubleValues: [Float!]
  floatValue: Float
  floatValues: [Float!]
  int32Value: Int
  int32Values: [Int!]
  int64Value: String
  int64Values: [String!]
  stringValue: String
  stringValues: [String!]
  timestamp: DateTime
  timestamps: [DateTime!]
  uint32Value: Int
  uint32Values: [Int!]
  uint64Value: String
  uint64Values: [String!]
}

type Query {
  test: Message!
}
"
`;

exports[`schema generation with ts-proto values are present 1`] = `
Object {
  "data": Object {
    "test": Object {
      "boolValue": true,
      "doubleValue": 2.4,
      "floatValue": 3.5,
      "int32Value": 2,
      "int64Value": "4",
      "stringValue": "foobar",
      "timestamp": "2020-12-28T06:42:05.453Z",
      "uint32Value": 5,
      "uint64Value": "6",
    },
  },
}
`;

exports[`schema generation with ts-proto values are present: schema.graphql 1`] = `
"\\"\\"\\"
A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the \`date-time\` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
\\"\\"\\"
scalar DateTime

type Message {
  boolValue: Boolean
  boolValues: [Boolean!]
  doubleValue: Float
  doubleValues: [Float!]
  floatValue: Float
  floatValues: [Float!]
  int32Value: Int
  int32Values: [Int!]
  int64Value: String
  int64Values: [String!]
  stringValue: String
  stringValues: [String!]
  timestamp: DateTime
  timestamps: [DateTime!]
  uint32Value: Int
  uint32Values: [Int!]
  uint64Value: String
  uint64Values: [String!]
}

input MessageInput {
  boolValue: Boolean
  boolValues: [Boolean!]
  doubleValue: Float
  doubleValues: [Float!]
  floatValue: Float
  floatValues: [Float!]
  int32Value: Int
  int32Values: [Int!]
  int64Value: String
  int64Values: [String!]
  stringValue: String
  stringValues: [String!]
  timestamp: DateTime
  timestamps: [DateTime!]
  uint32Value: Int
  uint32Values: [Int!]
  uint64Value: String
  uint64Values: [String!]
}

type Query {
  test: Message!
}
"
`;
